src/pages/api/proxy/[...path].ts

```diff
@@ -0,0 +1,53 @@
+import { EventEmitter } from 'events';
+
+import { createProxyMiddleware, fixRequestBody } from 'http-proxy-middleware';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+// Define error messages in a constant object
+const ERROR_MESSAGES = {
+  PROXY_ERROR: 'Proxy error',
+  PROXY_HANDLER_ERROR: 'Proxy handler error',
+};
+
+// Increase the max listeners to avoid memory leak
+EventEmitter.defaultMaxListeners = 20;
+
+const apiProxy = createProxyMiddleware<NextApiRequest, NextApiResponse>({
+  target: process.env.NEXT_PUBLIC_AUTH_BASE_URL,
+  changeOrigin: true,
+  pathRewrite: { '^/api/proxy': '' }, // eslint-disable-line @typescript-eslint/naming-convention
+  secure: false, // Disable SSL verification to avoid UNABLE_TO_VERIFY_LEAF_SIGNATURE error
```

Comment:
What are the implications of this and what happens when we have it set to `true?